src/providers/ApolloClientProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import {
+  ApolloClient,
+  ApolloProvider,
+  HttpLink,
+  InMemoryCache,
+  NormalizedCacheObject
+} from '@apollo/client'
+import { ConfigHelperConfig } from '@oceanprotocol/lib/dist/node/utils/ConfigHelper'
+import { useOcean } from '@oceanprotocol/react'
+import React, { useState, useEffect, ReactNode, ReactElement } from 'react'
+
+export default function ApolloClentProvider({
+  children
+}: {
+  children: ReactNode
+}): ReactElement {
+  const [client, setClient] = useState<ApolloClient<NormalizedCacheObject>>()
+  const { config } = useOcean()
+
+  useEffect(() => {
+    const newClient = new ApolloClient({
```

Comment:
good solution, putting it behind `useEffect` should make Gatsby not fail during its SSR build where it probably can't handle this client. To be safe, the effect could start with: 

```ts
useEffect(() => {
  if (!config?.subgraphUri) return
 
  const newClient = ...
}, [config]
```

Then we make sure it only ever fires once `config` and `config. subgraphUri` are available. This is a general pattern we have to follow whenever using values from `useOcean` and reacting to it. Because by default most of its exposed values are undefined and are filled during the lifecycle of the `OceanProvider`. So we always need to make sure we only do stuff once a value is not undefined anymore